### PR TITLE
Add NuGet download and pgk install in build.cmd

### LINKED ISSUE #165: First build fails (sign.target missing)
A target file is missing on first build (before the NuGet restore happens). Need to implement a bootstrapping step, or something equivalent, to get at least the BuildTools package first.

This is also causing CI builds to fail.

### PR DESCRIPTION
**Pull request by AndyGerlicher**
Closes #165

### CODE REVIEW

**Comment by rainersigwald** [comment]
At some point we should do this for xplat too, but it'll wait.

File: build.cmd [hunk 1]
```diff
@@ -13,6 +13,17 @@ if not defined MSBUILDCUSTOMPATH (
     set MSBUILDCUSTOMPATH=MSBuild.exe
 )
 
+set NUGETEXEPATH="%~dp0packages\NuGet.exe"
+
+if not exist "%NUGETEXEPATH%" (
+    :: This will need to be fixed for non-windows
+    echo ** Downloading NuGet.exe from https://nuget.org/NuGet.exe...
```
Comment (rainersigwald) [inline build.cmd]: Looks like you need to mkdir %~dp0packages

From the failed build:

** Downloading NuGet.exe from https://nuget.org/NuGet.exe...
PS> Invoke-WebRequest -OutFile "d:\j\workspace\microsoft_msbuild_prtest\packages\NuGet.exe" "https://nuget.org/NuGet.exe"
Invoke-WebRequest : Could not find a part of the path 
'D:\j\workspace\microsoft_msbuild_prtest\packages\NuGet.exe'.
At line:1 char:1
+ Invoke-WebRequest -OutFile 
d:\j\workspace\microsoft_msbuild_prtest\packages\NuGe ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~
    + CategoryInfo          : NotSpecified: (:) [Invoke-WebRequest], Directory 
   NotFoundException
    + FullyQualifiedErrorId : System.IO.DirectoryNotFoundException,Microsoft.P 
   owerShell.Commands.InvokeWebRequestCommand

**Comment by AndyGerlicher** [comment]
Build is green again!

**Comment by rainersigwald** [comment]
Woo! LGTM :+1:

**Pull request merged**